src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRateServiceTest.kt

```diff
@@ -412,16 +458,20 @@ class PayRateServiceTest {
 
     @Test
     fun `should not publish telemetry for no change`() {
-      val oldRates = listOf(
-        rateFactory().builder(endDate = yesterday.toString()).create(courseActivity),
-        rateFactory().builder(startDate = tomorrow.toString(), halfDayRate = 4.3).create(courseActivity),
-        rateFactory().builder(payBandCode = "7", halfDayRate = 8.7).create(courseActivity),
-      )
-      val newRates = listOf(
-        rateFactory().builder(endDate = yesterday.toString()).create(courseActivity),
-        rateFactory().builder(startDate = tomorrow.toString(), halfDayRate = 4.3).create(courseActivity),
-        rateFactory().builder(payBandCode = "7", halfDayRate = 8.7).create(courseActivity),
-      )
+      val newRates = mutableListOf<CourseActivityPayRate>()
+      nomisDataBuilder.build {
+        programService {
+          courseActivity {
+            newRates.addAll(
+              listOf(
+                payRate(endDate = yesterday.toString()),
+                payRate(startDate = tomorrow.toString(), halfDayRate = 4.3),
+                payRate(payBandCode = "7", halfDayRate = 8.7),
+              ),
+            )
+          }
+        }
+      }
 
       val telemetry = payRatesService.buildUpdateTelemetry(oldRates, newRates)
```

Comment:
Here's an example of using the DSL generated entities in a unit test.
`newRates.addAll(` takes collects the results of all `payRate` calls and to be used later in the test.